### PR TITLE
Scm sources

### PR DESCRIPTION
**Pull request by adrianschroeter**
Some important hints how to work with SCM sources.

I know that the book build is broken, but I have no clue why, since the reported false string ("format") does not exist in the file.

Any help is welcome:) Either a hint or just fix it yourself, thanks!

### CODE REVIEW

File: xml/obs_best_practice_integrate_scm_sources.xml [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+     OBS will always keep a copy of the sources to guarantee that the build sources
+     are still available even when the external SCM server disappears or get altered.
+   </para>
+   <sect1>
+     <title>How to create a source service</title>
+     <para>
+      Let OBS create a tar ball out of an SCM repository. This just creates or
+      extend a _service file with some rules how to download and package sources.
+      The actual work happens on a local build or on a service side build. Please
+      note that you need the obs-service-obs_scm installed for local runs.
```
Comment (fsundermeyer) [inline xml/obs_best_practice_integrate_scm_sources.xml]: </para>

File: xml/obs_best_practice_integrate_scm_sources.xml [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+      <screen>
+        <command>
+          osc add https://SOME_URL.git
+        </command>
+      </screen>
```
Comment (fsundermeyer) [inline xml/obs_best_practice_integrate_scm_sources.xml]: <para>

File: xml/obs_best_practice_integrate_scm_sources.xml [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+      The web interface is creating as well a _service file when adding an URL
+      to a SCM system.
+     </para>
+     <sect2>
+       <title>Follow upstream branches</title>
+       <para>
+       The created _service file is set up to follow latest source submissions on each
+       run and looks like this:
```
Comment (fsundermeyer) [inline xml/obs_best_practice_integrate_scm_sources.xml]: </para>

File: xml/obs_best_practice_integrate_scm_sources.xml [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+		  &lt;service name="set_version" mode="buildtime"/&gt;
+		  &lt;service name="tar" mode="buildtime"/&gt;
+		  &lt;service name="recompress" mode="buildtime"&gt;
+		    &lt;param name="file"&gt;*.tar&lt;/param&gt;
+		    &lt;param name="compression"&gt;xz&lt;/param&gt;
+		  &lt;/service&gt;
+		&lt;/services&gt;
+          </command>
+        </screen>
```
Comment (fsundermeyer) [inline xml/obs_best_practice_integrate_scm_sources.xml]: <para>

File: xml/obs_best_practice_integrate_scm_sources.xml [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        To follow a specific branch and additional parameter for "revision" is needed for the
+        obs_scm service.
+       </para>
+     </sect2>
+     <sect2>
+       <title>Fixed versions</title>
+       <para>
+         You may want to build an archive for a fixed version, for example an official release
+         which has been tagged by the upstream project. It is recommend to specify the 
+         mode="disabled" and to submit the archive via the following
```
Comment (fsundermeyer) [inline xml/obs_best_practice_integrate_scm_sources.xml]: commands: </para>

File: xml/obs_best_practice_integrate_scm_sources.xml [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+        <screen>
+          <command>
+	    # osc service runall
+	    # osc ar
+	    # osc commit
+          </command>
+        </screen>
+        commands.
+       </para>
```
Comment (fsundermeyer) [inline xml/obs_best_practice_integrate_scm_sources.xml]: Remove these lines

78       commands.
￼79       &lt;/para&gt;

File: xml/obs_best_practice_integrate_scm_sources.xml [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+     <sect2>
+       <title>Avoid tar balls</title>
+       <para>
+         tar balls are not a requirement by &obsa;, but by the packaging tool, eg. rpmbuild. However,
+         you may want to decide not to ship a tar ball inside of the src.rpm. This makes sense for
+         large sources where the compression time and needed disk space is just considered a waste
+         for short living builds. You can simplify your _service file in that case, but you need to help rpmbuild to work directly in the source.
+         It is also a good practice to package the _service file instead of the tar ball to 
+         give the user a chance to rebuild the src.rpm as long the external SCM server is
+         providing the sources. The simplified _service file looks like this:
```
Comment (fsundermeyer) [inline xml/obs_best_practice_integrate_scm_sources.xml]: </para>

File: xml/obs_best_practice_integrate_scm_sources.xml [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+            &lt;services&gt;
+		  &lt;service name="obs_scm"&gt;
+		    &lt;param name="url"&gt;https://github.com/FreeCAD/FreeCAD.git&lt;/param&gt;
+		    &lt;param name="scm"&gt;git&lt;/param&gt;
+		  &lt;/service&gt;
+
+		  &lt;service name="set_version" mode="buildtime"/&gt;
+            &lt;/services&gt;
+          </command>
+        </screen>
```
Comment (fsundermeyer) [inline xml/obs_best_practice_integrate_scm_sources.xml]: <para>

File: xml/obs_best_practice_integrate_scm_sources.xml [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+        The spec file needs some hints to build inside the extracted sources directly. The macro
+        can be used to switch to build tar balls or not to keep it working for stable releases where
+        you want to provide a complete source rpm.
```
Comment (fsundermeyer) [inline xml/obs_best_practice_integrate_scm_sources.xml]: </para>

File: xml/obs_best_practice_integrate_scm_sources.xml [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+             ...
+             %prep
+             %if %{build_tar_ball}
+              %setup -q
+             %else
+              %setup -q -n %_sourcedir/%name-%version -T -D
+             %endif
+          </command>
+        </screen>
+       </para>
```
Comment (fsundermeyer) [inline xml/obs_best_practice_integrate_scm_sources.xml]: Remove this line

125 &lt;/para&gt;

File: xml/obs_best_practice_integrate_scm_sources.xml [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+   <sect1>
+     <title>How to create a source service</title>
+     <para>
+      Let OBS create a tar ball out of an SCM repository. This just creates or
+      extend a _service file with some rules how to download and package sources.
+      The actual work happens on a local build or on a service side build. Please
+      note that you need the obs-service-obs_scm installed for local runs.
+      <screen>
+        <command>
+          osc add https://SOME_URL.git
```
Comment (fsundermeyer) [inline xml/obs_best_practice_integrate_scm_sources.xml]: Remove whitespace at beginning of line

File: xml/obs_best_practice_integrate_scm_sources.xml [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+      to a SCM system.
+     </para>
+     <sect2>
+       <title>Follow upstream branches</title>
+       <para>
+       The created _service file is set up to follow latest source submissions on each
+       run and looks like this:
+        <screen>
+          <command>
+            &lt;services&gt;
```
Comment (fsundermeyer) [inline xml/obs_best_practice_integrate_scm_sources.xml]: Wrong indent

File: xml/obs_best_practice_integrate_scm_sources.xml [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+       <para>
+         You may want to build an archive for a fixed version, for example an official release
+         which has been tagged by the upstream project. It is recommend to specify the 
+         mode="disabled" and to submit the archive via the following
+        <screen>
+          <command>
+	    # osc service runall
+	    # osc ar
+	    # osc commit
+          </command>
```
Comment (fsundermeyer) [inline xml/obs_best_practice_integrate_scm_sources.xml]: Remove whitespace at beginning of lines

File: xml/obs_best_practice_integrate_scm_sources.xml [hunk 14]
```diff
@@ -0,0 +1,128 @@
...
+      <screen>
+        <command>
+          osc add https://SOME_URL.git
+        </command>
+      </screen>
+      The web interface is creating as well a _service file when adding an URL
+      to a SCM system.
+     </para>
+     <sect2>
+       <title>Follow upstream branches</title>
```
Comment (tomschr) [inline xml/obs_best_practice_integrate_scm_sources.xml]: Use title style: Following Upstream Branches

File: xml/obs_best_practice_integrate_scm_sources.xml [hunk 15]
```diff
@@ -0,0 +1,128 @@
...
+       <para>
+       The created _service file is set up to follow latest source submissions on each
```
Comment (tomschr) [inline xml/obs_best_practice_integrate_scm_sources.xml]: Use `filename` around `_service`

File: xml/obs_best_practice_integrate_scm_sources.xml [hunk 16]
```diff
@@ -0,0 +1,128 @@
...
+	    # osc service runall
+	    # osc ar
+	    # osc commit
+          </command>
+        </screen>
+        commands.
+       </para>
+     </sect2>
+     <sect2>
+       <title>Avoid tar balls</title>
```
Comment (tomschr) [inline xml/obs_best_practice_integrate_scm_sources.xml]: To be consistent. use (and avoid "balls"): "Avoiding Tar Archive"

File: xml/obs_best_practice_integrate_scm_sources.xml [hunk 17]
```diff
@@ -0,0 +1,128 @@
...
+       <para>
+         tar balls are not a requirement by &obsa;, but by the packaging tool, eg. rpmbuild. However,
```
Comment (tomschr) [inline xml/obs_best_practice_integrate_scm_sources.xml]: "Tar archives ..."

File: xml/obs_best_practice_integrate_scm_sources.xml [hunk 18]
```diff
@@ -0,0 +1,128 @@
+<!DOCTYPE chapter PUBLIC
+  "-//OASIS//DTD DocBook XML V4.5//EN"
+  "http://www.docbook.org/xml/4.5/docbookx.dtd"
+[
+  <!ENTITY % entities SYSTEM "entity-decl.ent">
+  %entities;
+]>
```
Comment (tomschr) [inline xml/obs_best_practice_integrate_scm_sources.xml]: That DOCTYPE is wrong in the context of DocBook 5. It must be:

```xml
<!DOCTYPE chapter [
  <!ENTITY % entities SYSTEM "entity-decl.ent">
  %entities;
]>
```

File: xml/obs_best_practice_integrate_scm_sources.xml [hunk 19]
```diff
@@ -0,0 +1,128 @@
...
+      Let OBS create a tar ball out of an SCM repository. This just creates or
+      extend a _service file with some rules how to download and package sources.
+      The actual work happens on a local build or on a service side build. Please
+      note that you need the obs-service-obs_scm installed for local runs.
+      <screen>
+        <command>
+          osc add https://SOME_URL.git
+        </command>
+      </screen>
+      The web interface is creating as well a _service file when adding an URL
```
Comment (tomschr) [inline xml/obs_best_practice_integrate_scm_sources.xml]: Add `filename` around `_service`.

File: xml/obs_best_practice_integrate_scm_sources.xml [hunk 20]
```diff
@@ -0,0 +1,128 @@
...
+		  &lt;service name="tar" mode="buildtime"/&gt;
+		  &lt;service name="recompress" mode="buildtime"&gt;
+		    &lt;param name="file"&gt;*.tar&lt;/param&gt;
+		    &lt;param name="compression"&gt;xz&lt;/param&gt;
+		  &lt;/service&gt;
+		&lt;/services&gt;
+          </command>
+        </screen>
+        This will create an obscpio archive via the obs_scm service with the latest sources. 
+        This archive will get extracted at build time and be processed via the other services
```
Comment (tomschr) [inline xml/obs_best_practice_integrate_scm_sources.xml]: What do you mean by "other services"`?

File: xml/obs_best_practice_integrate_scm_sources.xml [hunk 21]
```diff
@@ -0,0 +1,128 @@
...
+        to build a compressed tar ball for rpmbuild. 
+        To follow a specific branch and additional parameter for "revision" is needed for the
+        obs_scm service.
+       </para>
+     </sect2>
+     <sect2>
+       <title>Fixed versions</title>
```
Comment (tomschr) [inline xml/obs_best_practice_integrate_scm_sources.xml]: Style and consistency: "Fixing Versions"

File: xml/obs_best_practice_integrate_scm_sources.xml [hunk 22]
```diff
@@ -0,0 +1,128 @@
...
+       <para>
+         You may want to build an archive for a fixed version, for example an official release
```
Comment (tomschr) [inline xml/obs_best_practice_integrate_scm_sources.xml]: * missing comma: for example an -> for example, an

File: xml/obs_best_practice_integrate_scm_sources.xml (hunk 5 above)
Comment (tomschr) [inline xml/obs_best_practice_integrate_scm_sources.xml]: How to set this mode? Where?

File: xml/obs_best_practice_integrate_scm_sources.xml [hunk 23]
```diff
@@ -0,0 +1,128 @@
...
+
+		  &lt;service name="set_version" mode="buildtime"/&gt;
+            &lt;/services&gt;
+          </command>
+        </screen>
+        The spec file needs some hints to build inside the extracted sources directly. The macro
+        can be used to switch to build tar balls or not to keep it working for stable releases where
+        you want to provide a complete source rpm.
+        <screen>
+          <command>
```
Comment (tomschr) [inline xml/obs_best_practice_integrate_scm_sources.xml]: * Remove `command`.
* Remove any indendation, start content of `screen` always at first column.

**Pull request merged**